### PR TITLE
Add API Access Token option to Stack Overflow backend plugin

### PR DESCRIPTION
**Pull request by ConnorDY**
## Hey, I just made a Pull Request!

     
In order to use this plugin with a private Stack Overflow Team, the `X-API-Access-Token` header must be set.

Then requests like the following will work:
```
https://api.stackexchange.com/2.2/questions?pagesize=100&site=stackoverflow&team=<team>&key=<key>
```

Without the header, the following error occurs:
```json
{
    "error_id": 403,
    "error_message": "Access token is required to query Team specific data",
    "error_name": "access_denied"
}
```

#### :heavy_check_mark: Checklist

- [x] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [x] Added or updated documentation
- [ ] Tests for new functionality and regression tests for bug fixes
- [ ] Screenshots attached (for UI changes)
- [x] All your commits have a `Signed-off-by` line in the message. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))

### CODE REVIEW

**Review by iamEAP — COMMENTED** [review]
Thanks for the contribution @ConnorDY!

I'm reading through some of the Stackoverflow docs related to this, and the impression I'm getting is that the `X-API-Access-Token` header is intended to be used in combination with some kind of OAuth flow.  ...However the SOE collator is more of a server-to-server integration where OAuth isn't really possible.

Could you help me understand what you're trying to do?

**Comment by ConnorDY** [comment]
> I'm reading through some of the Stackoverflow docs related to this, and the impression I'm getting is that the X-API-Access-Token header is intended to be used in combination with some kind of OAuth flow.

After setting up an OAuth App on Stack Apps, you can use that OAuth App to generate an API Access Token with no expiration. This is explained in the docs [here](https://stackoverflow.help/en/articles/4385859-stack-overflow-for-teams-api) (Step 4: "Generate an Access Token via OAuth").

So the collator does not require any code to handle the OAuth flow. This is similar to using a GitHub PAT associated with a service account.

I have tested these code changes in my team's Backstage repo and can verify that this works.

File: plugins/stack-overflow-backend/README.md [hunk 1]
```diff
@@ -23,6 +23,7 @@ If you have a private stack overflow instance you will need to supply an API key
 stackoverflow:
   baseUrl: https://api.stackexchange.com/2.2 # alternative: your internal stack overflow instance
   apiKey: $STACK_OVERFLOW_API_KEY
+  apiAccessToken: $STACK_OVERFLOW_API_ACCESS_TOKEN
```
Comment (iamEAP) [inline plugins/stack-overflow-backend/README.md:26]: In addition to documenting this here, you'll also want to add it to `config.d.ts` with `@visibility secret` ([see](https://github.com/backstage/backstage/blob/6756292c2499ea6de1f96dffba442c056120bd92/plugins/stack-overflow-backend/config.d.ts#L27-L31)) <-- While you're in there, it'd be good to differentiate the two API config entries in the doc blocks there too.

Comment (iamEAP) [reply plugins/stack-overflow-backend/README.md:26]: A secondary thought to this: is usage of the API Key vs. this access token exclusive (e.g. would you ever use both simultaneously)?  If they're exclusive, you might also play with the typescript so that only one is allowed to be defined at any given time.

File: plugins/stack-overflow-backend/README.md [hunk 2]
```diff
@@ -23,6 +23,7 @@ If you have a private stack overflow instance you will need to supply an API key
 stackoverflow:
```
Comment (iamEAP) [inline plugins/stack-overflow-backend/README.md:23]: Could be good to call out the documentation you linked in this PR in the section above (so adopters know how to get the value)

**Review by iamEAP — COMMENTED** [review]
Thanks for the clarification @ConnorDY.  In that case, changes look good!  I have a few comments, mostly around documentation, with the `config.d.ts` being the most critical.

If you _really_ wanted to, adding tests for the collator could be nice!  But given there are none currently, no sweat if you don't have time.  Check out the [TestPipeline](https://backstage.io/docs/reference/plugin-search-backend-node.testpipeline) class as well as other collator tests in the repo for inspiration.

**Review by iamEAP — APPROVED** [review]
